Source/Utils/WarpXUtil.H

```diff
@@ -182,6 +182,15 @@ safeCastToInt(amrex::Real x, const std::string& real_name);
 */
 amrex::Parser makeParser (std::string const& parse_function, amrex::Vector<std::string> const& varnames);
 
+/**
+ * \brief Parse a string and return either a real or an int
+ *
+ * \param str The string to be parsed
+ * \param name For integers, the name, to be used in error messages
+ */
+amrex::Real parseStringtoReal(std::string str);
+int parseStringtoInt(std::string str, std::string name);
```

Comment:
Technically to parse this well, we unfortunately need to do thisL
```suggestion
/** Parse a string and return as a real
 *
 * In case the string cannot be interpreted as Real,
 * this function ... <throws an exception? aborts with error message?>
 *
 * \param str The string to be parsed
 * \return representation as real
 */
amrex::Real parseStringtoReal(std::string str);

/** Parse a string and return an int
 *
 * In case the string cannot be interpreted as Real,
 * this function ... <throws an exception? aborts with error message?>
  *
 * \param str The string to be parsed
 * \param name For integers, the name, to be used in error messages
 * \return rounded closest integer
 */
int parseStringtoInt(std::string str, std::string name);
```

Reply:
Nit pick: I would probably spell `parseStringtoInt` as `parseStringToInt` and
`parseStringtoReal` as `parseStringToReal`